### DISCUSSION TITLE
tryit

### DISCUSSION THREAD

**Discussion by azsembler** [rayepps/radash#94] [General]
Hi, I used tryit(parallel) function to expect both result and error but the error return as type Error at new AggregateError.
When I try to print out err.errors, my IDE said err is type Error and it has only message name and stack.

I have no idea how to get the errors values like the document

Reference -> https://radash-docs.vercel.app/docs/async-parallel

**Comment by sodiray** [comment]
Hey @azsembler this is a weakness of Typescript unfortunately 😞 It's not possible to type the error that is thrown and/or pass it as a generic so the `try` function has to do a _best effort_ typing and return `[Error, Result]`. You and I (because we're smarter than the Typescript compiler) know that when using `try` with `parallel` it's actually going to throw an `AggregateError` instead of an `Error` but theres no way to communicate that to Typescript (that I'm aware of) besides manually overriding it.

You'll have to do a bit of type trickery to tell your IDE and Typescript that the error you get back is of type `AggregateError`, not `Error`.

<img width="1300" alt="Screen Shot 2022-10-13 at 1 19 02 PM" src="https://user-images.githubusercontent.com/15269623/195701879-5a058a59-4cb0-4987-a0b3-42ba032856bb.png">

If anyone knows a better way, I'd love to hear it! This is what I have to do in this case.
